Nodejs/Product/Nodejs/Project/NodeModulesNode.cs

```diff
@@ -130,20 +136,20 @@ private class NpmPathProvider : INpmPathProvider {
             }
         }
 
-        private INpmController CreateNpmController() {
-            if (null == _npmController) {
-                _npmController = NpmControllerFactory.Create(
-                    _projectNode.ProjectHome,
-                    NodejsPackage.Instance.NpmOptionsPage.NpmCachePath,
-                    false,
-                    new NpmPathProvider(this));
-                _npmController.CommandStarted += NpmController_CommandStarted;
-                _npmController.OutputLogged += NpmController_OutputLogged;
-                _npmController.ErrorLogged += NpmController_ErrorLogged;
-                _npmController.ExceptionLogged += NpmController_ExceptionLogged;
-                _npmController.CommandCompleted += NpmController_CommandCompleted;
-            }
-            return _npmController;
+        private static INpmController DefaultNpmController(string projectHome, NpmPathProvider pathProvider) {
+            return NpmControllerFactory.Create(
+                projectHome,
+                NodejsPackage.Instance.NpmOptionsPage.NpmCachePath,
+                false,
+                pathProvider);
+        }
+
+        private void RegisterWithNpmController(INpmController controller) {
+            controller.CommandStarted += NpmController_CommandStarted;
+            controller.OutputLogged += NpmController_OutputLogged;
+            controller.ErrorLogged += NpmController_ErrorLogged;
+            controller.ExceptionLogged += NpmController_ExceptionLogged;
+            controller.CommandCompleted += NpmController_CommandCompleted;
```

Comment:
That makes sense, but we should only optimize for patterns if we're actually using them. Otherwise we get into situations where we have to keep hoping around the code in order to actually get to the heart of what it's doing. Right now this code is